content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+
+```csharp
+    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+    dynamic data = JsonConvert.DeserializeObject(requestBody);
+    string pageViewURL = data?.URL;
+```
+
+And now, we can use `URL` to create a `ViewCount` object that uses it as a key. The logic should be easy to follow but here's a high-level overview:
+
+1. Try to retrieve the `ViewCount` object for the URL.
+1. If it doesn't exist, create a new `ViewCount` object (with count 0 by default) for that URL.
+1. Increase the view count on that object by 1.
+1. Store the updated entity.
+1. Return the view count to the function caller.
+
+Here's what the code looks like:
+
+```csharp
+     var retrievedResult = table.Execute(TableOperation.Retrieve<ViewCount>(pageViewURL, "visits")); // try to retrieve count for the URL
+     var pageViewCount = (ViewCount) retrievedResult.Result; // wait for the operation and get the result into a ViewCount object 
+  
+     pageViewCount = pageViewCount ?? new ViewCount(pageViewURL); // if the entity didn't exist, create a new one with count 0
+     pageViewCount.Count++; // augment the view count by one
+     
+     table.Execute(TableOperation.InsertOrReplace(pageViewCount)); // insert or replace the entity with the new count
+     return (ActionResult)new OkObjectResult(pageViewCount.Count.ToString()); // return the view count 
+```
+
+Let's also make sure we return an error code if we don't give a "URL" identifier to the function before trying to retrieve an entity:
+
+```csharp
+  if (pageViewURL == null)
+  {
+    return (ActionResult)new StatusCodeResult(503);
+  }
+```
+The function is now ready! Let's make sure it works by running it locally and performing POST requests that have a "URL" parameter in the body.
+
+### Deploying and implementation
+
+We are now ready to deploy the Azure Function. Something that you might want to make sure of, is that you do not usually want to keep any connection string in code, so we will be getting it through the [App Settings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings#settings):
+
+```csharp
+  // we will have a StorageConnectionString field in the app settings where we store the storage account connection string
+  var storageAccountConnectionString = Environment.GetEnvironmentVariable("StorageConnectionString");
+```
+The App Settings can be changed in the Azure Portal at runtime.  
+
+Lastly, you can implement this on any page by creating a small script that calls into this function and gets the result:
+```js
+var url = 'https://pageviewcounterblog.azurewebsites.net/api/GetViewCounter/';
+var data = {URL: 'this-blog-post-URL'};
+
+fetch(url, {
+  method: 'POST', 
+  body: JSON.stringify(data), 
+  headers:{
+    'Content-Type': 'application/json'
+  }
+}).then(res => res.json())
+.then(response => console.log('Success:', JSON.stringify(response)))
+```
```

Comment:
How about

```js
const url = 'https://pageviewcounterblog.azurewebsites.net/api/GetViewCounter/';
const data = {URL: 'this-blog-post-URL'};

const response = await fetch(url, {
  method: 'POST', 
  body: JSON.stringify(data), 
  headers: {
    'Content-Type': 'application/json'
  }
})
const body = await response.json()
console.log('Success:', JSON.stringify(response))
```

Reply:
Don't have a strong opinion for those (I just searched for the 'common' way to do POST requests with JS), so I will use your solution since you are the expert.